pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -492,3 +511,15 @@ func (ctx *context) handleInboundResponse(response *Response) (stateAction, erro
 func getEpochTime() int64 {
 	return time.Now().Unix()
 }
+func getPublicKey(didDoc *did.Doc, pubKeyType string) ([]did.PublicKey, error) {
+	var publicKeys []did.PublicKey
+	for k := range didDoc.PublicKey {
+		if didDoc.PublicKey[k].Type == pubKeyType {
```

Comment:
why not have range return the pubKey rather than looking up again?

Reply:
done